src/OrleansSQLUtils/Storage/Provider/RelationalStorageProvider.cs

```diff
@@ -0,0 +1,550 @@
+            if(string.IsNullOrWhiteSpace(name))
+            {
+                throw new ArgumentException("The parameter must contain characters", nameof(name));
+            }
+
+            if(providerRuntime == null)
+            {
+                throw new ArgumentNullException(nameof(providerRuntime));
+            }
+
+            if(config == null)
+            {
+                throw new ArgumentNullException(nameof(config));
+            }
+
+            if(!config.Properties.ContainsKey(DataConnectionStringPropertyName))
+            {
+                throw new BadProviderConfigException($"The {DataConnectionStringPropertyName} setting has not been configured. Add a {DataConnectionStringPropertyName} setting with a valid connection string.");
+            }
+
+            //NOTE: StorageSerializationPicker should be defined outside and given as a parameter in constructor or via Init in IProviderConfiguration perhaps.
+            //Currently this limits one's options to much to the current situation of providing only one serializer for serialization and deserialization
+            //with no regard to state update or serializer changes. Maybe have this serialized as a JSON in props and read via a key?
+            StorageSerializationPicker = new DefaultRelationalStoragePicker(ConfigureDeserializers(config), ConfigureSerializers(config));
+
+            //NOTE: Currently there should be only one pair of providers given. That is, only UseJsonFormatPropertyName, UseXmlFormatPropertyName or UseBinaryFormatPropertyName.
+            if(StorageSerializationPicker.Deserializers.Count > 1 || StorageSerializationPicker.Serializers.Count > 1)
+            {
+                throw new ArgumentException("Configuration error, only one serializer and deserializer should be given.", nameof(config));
+            }
+
+            if(StorageSerializationPicker.Deserializers.Count == 0 || StorageSerializationPicker.Serializers.Count == 0)
+            {
+                StorageSerializationPicker.Deserializers.Add(new OrleansStorageDefaultBinaryDeserializer(UseBinaryFormatPropertyName));
+                StorageSerializationPicker.Serializers.Add(new OrleansStorageDefaultBinarySerializer(UseBinaryFormatPropertyName));
+            }
+
+            var connectionInvariant = config.Properties.ContainsKey(DataConnectionInvariantPropertyName) ? config.Properties[DataConnectionInvariantPropertyName] : DefaultAdoInvariantInvariantPropertyName;
+            Storage = RelationalStorage.CreateInstance(connectionInvariant, config.Properties[DataConnectionStringPropertyName]);
+            ServiceId = providerRuntime.ServiceId.ToString();
+
+            var queries = await Storage.ReadAsync(DefaultInitializationQuery, command => { }, (selector, resultSetCount, token) =>
+            {
+                return Task.FromResult(Tuple.Create(selector.GetValue<string>("QueryKey"), selector.GetValue<string>("QueryText")));
+            }).ConfigureAwait(false);
```

Comment:
I think that the usage `ConfigureAwait(false)` here and in other places in this PR is wrong. The task continuation should be on the captured context. No one should call the code using `Wait()` or `Result` or `GetAwaiter().GetResult()`.